server/src/main/scala/org/scassandra/server/ServerStubRunner.scala

```diff
@@ -52,42 +56,32 @@ class ServerStubRunner( val binaryListenAddress: String = "localhost",
                         val adminListenAddress: String = "localhost",
                         val adminPortNumber: Int = 8043,
                         val startupTimeoutSeconds: Long = 10) extends LazyLogging {
+  import ServerStubRunner.actorSystem
 
-  var system: ActorSystem = _
+  import ExecutionContext.Implicits.global
 
+  // TODO: This is only used by integration tests, move into Actor.
   val primedResults = PrimeQueryStore()
-  val primePreparedStore = new PrimePreparedStore
-  val primePreparedPatternStore = new PrimePreparedPatternStore
-  val primePreparedMultiStore = new PrimePreparedMultiStore
-  val primeBatchStore = new PrimeBatchStore
-  val activityLog = new ActivityLog
-  val preparedLookup = new CompositePreparedPrimeStore(primePreparedStore, primePreparedPatternStore, primePreparedMultiStore)
-
-  var primingReadyListener: ActorRef = _
-  var tcpReadyListener: ActorRef = _
-
-  def start() = {
-    system = ActorSystem(s"CassandraServerStub-$binaryPortNumber-$adminPortNumber")
-    val manager = IO(Tcp)(system)
-    primingReadyListener = system.actorOf(Props(classOf[ServerReadyListener]), "PrimingReadyListener")
-    tcpReadyListener = system.actorOf(Props(classOf[ServerReadyListener]), "TcpReadyListener")
-    val tcpServer = system.actorOf(Props(classOf[TcpServer], manager, binaryListenAddress, binaryPortNumber, primedResults, preparedLookup, primeBatchStore, tcpReadyListener, activityLog), "BinaryTcpListener")
-    system.actorOf(Props(classOf[PrimingServer], adminListenAddress, adminPortNumber, primedResults, primePreparedStore,
-      primePreparedPatternStore, primePreparedMultiStore, primeBatchStore, primingReadyListener, activityLog, tcpServer), "PrimingServer")
-  }
 
-  def awaitTermination() = {
-    system.awaitTermination()
+  var scassandra: ActorRef = _
+
+  def start() = this.synchronized {
+    scassandra = actorSystem.actorOf(Props(classOf[ScassandraServer], primedResults, binaryListenAddress,
+      binaryPortNumber, adminListenAddress, adminPortNumber))
   }
 
-  def shutdown() = {
-    system.shutdown()
-    system.awaitTermination()
-    logger.info("Server is shut down")
+  def shutdown() = this.synchronized {
+    val stopped = gracefulStop(scassandra, startupTimeoutSeconds + 1 seconds, ShutdownServer(startupTimeoutSeconds seconds))
```

Comment:
Unfortunately using a PoisonPill here was not enough, it seemed that the actors created by the IO managers would not be guaranteed to be killed to unbind their listening interfaces when their listener actors stopped, so instead I created a `ShutdownServer` message that causes the `PrimingServer` and `TcpServer` actors to first send an `Unbind` to their IO listeners, wait for Unbound and then send a `PoisonPill` to themselves as part of a shutdown process.  This ensures any listening interfaces die before the actors die.  This wasn't needed previously as the entire `ActorSystem` was shut down as part of the process.